api/src/main/java/org/pmiops/workbench/initialcredits/InitialCreditsService.java

```diff
@@ -368,8 +368,24 @@ public boolean areUserCreditsExpired(DbUser user) {
         .orElse(false);
   }
 
-  // Returns true if the user's credits are expiring within the initialCreditsExpirationWarningDays.
+  private boolean checkEnabledInitialCreditsExpirationFlag() {
+    return workbenchConfigProvider.get().featureFlags.enableInitialCreditsExpiration;
```

Comment:
As we discussed, this should check `enableUnlinkBillingForInitialCredits` instead